examples/ecs/async_ecs.rs

```diff
@@ -0,0 +1,144 @@
+//! A minimal example showing how to perform asynchronous work in Bevy
+//! using [`AsyncComputeTaskPool`] for parallel task execution and a crossbeam channel
+//! to communicate between async tasks and the main ECS thread.
+//!
+//! This example demonstrates how to spawn detached async tasks, send completion messages via channels,
+//! and dynamically spawn ECS entities (cubes) as results from these tasks. The system processes
+//! async task results in the main game loop, all without blocking or polling the main thread.
+
+use bevy::{
+    math::ops::{cos, sin},
+    prelude::*,
+    tasks::AsyncComputeTaskPool,
+};
+use futures_timer::Delay;
+use rand::Rng;
+use std::time::Duration;
+
+const NUM_CUBES: i32 = 16;
+const LIGHT_RADIUS: f32 = 8.0;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(
+            Startup,
+            (setup_env, setup_assets, spawn_tasks.after(setup_assets)),
+        )
+        .add_systems(Update, rotate_light)
+        .run();
+}
+
+/// Spawns async tasks on the compute task pool to simulate delayed cube creation.
+///
+/// Each task is executed on a separate thread and sends the result (cube position)
+/// back through the `CubeChannel` once completed. The tasks are detached to
+/// run asynchronously without blocking the main thread.
+///
+/// In this example, we don't implement task tracking or proper error handling.
+fn spawn_tasks(world_id: WorldId) {
+    let pool = AsyncComputeTaskPool::get();
+    let task_id = EcsTask::new(world_id);
+    for x in -NUM_CUBES..NUM_CUBES {
+        for z in -NUM_CUBES..NUM_CUBES {
+            // Spawn a task on the async compute pool
+            pool.spawn(async move {
+                let delay = Duration::from_secs_f32(rand::rng().random_range(2.0..8.0));
+                // Simulate a delay before task completion
+                println!("delaying for {:?}", delay);
+                Delay::new(delay).await;
+                if let Err(e) = async_access::<
```

Comment:
Is it possible to rewrite this API to not require the wildcard generic parameters? Or are those unavoidable? Or maybe just set a standard for adding these params at the `EcsTask::new` site? Or annotating the closure's parameter types instead of the generic?